test/Confluent.Kafka.Avro.IntegrationTests/Tests/AvroAndRegular.cs

```diff
@@ -0,0 +1,146 @@
+
+                // check that the value type was registered with SR, and the key was not.
+                Assert.Throws<SchemaRegistryException>(() =>
+                    {
+                        try
+                        {
+                            schemaRegistry.GetLatestSchemaAsync(schemaRegistry.ConstructKeySubjectName(topic1)).Wait();
+                        }
+                        catch (AggregateException e)
+                        {
+                            throw e.InnerException;
+                        }
+                    });
+                var s2 = schemaRegistry.GetLatestSchemaAsync(schemaRegistry.ConstructValueSubjectName(topic1)).Result;
+            }
+
+            using (var schemaRegistry = new CachedSchemaRegistryClient(schemaRegistryConfig))
+            using (var producer = new AvroProducer(producerConfig))
+            {
+                producer.RegisterAvroSerializer(new AvroSerializer<string>(schemaRegistry));
+
+                // implicit check that this does not fail.
+                producer.ProduceAsync(topic2, new Message<string, string> { Key = "hello", Value = "world" }, SerdeType.Avro, SerdeType.Regular).Wait();
+
+                // check that the key type was registered with SR, and the value was not.
+                Assert.Throws<SchemaRegistryException>(() =>
+                    {
+                        try
+                        {
+                            schemaRegistry.GetLatestSchemaAsync(schemaRegistry.ConstructValueSubjectName(topic2)).Wait();
+                        }
+                        catch (AggregateException e)
+                        {
+                            throw e.InnerException;
+                        }
+                    });
+                var s2 = schemaRegistry.GetLatestSchemaAsync(schemaRegistry.ConstructKeySubjectName(topic2)).Result;
+            }
+
+            // check the above can be consumed (using regular / avro serializers as appropriate)
+            using (var schemaRegistry = new CachedSchemaRegistryClient(schemaRegistryConfig))
+            using (var consumer = new AvroConsumer(consumerConfig))
+            {
+                consumer.RegisterAvroDeserializer<string>(new AvroDeserializer<string>(schemaRegistry));
+
+                consumer.Assign(new TopicPartitionOffset(topic1, 0, 0));
+                var cr = consumer.ConsumeAsync<string, string>(SerdeType.Regular, SerdeType.Avro).Result;
+                Assert.Equal("hello", cr.Key);
+                Assert.Equal("world", cr.Value);
+
+                consumer.Assign(new TopicPartitionOffset(topic2, 0, 0));
+                cr = consumer.ConsumeAsync<string, string>(SerdeType.Avro, SerdeType.Regular).Result;
+                Assert.Equal("hello", cr.Key);
+                Assert.Equal("world", cr.Value);
+
+                // TODO: review potential exceptions.
```

Comment:
this needs reviewing before 1.0 is released. two things - it throws a doubly nested `AggregateException` with inner exception `InvalidDataException` - probably want something avro specific. have noted this.

Reply:
fixed - all serdes now throw `DeserializationException` appropriately, wrapping the exception indicating whatever the problem was. also, nested AggregationExceptions are now unnested.